src/pages/index.js

```diff
@@ -82,31 +142,82 @@ const editFormValidator = new FormValidator(
 );
 const addFormValidator = new FormValidator(validationSettings, addCardModal);
 
+const editAvatarValidator = new FormValidator(
+  validationSettings,
+  avatarEditModal
+);
+
 //Initialize all instances
-cardSection.renderItems(initialCards);
+api.getInitialCards().then((cards) => {
+  cardSection.renderItems(cards);
+});
+
 cardPreviewPopup.setEventListeners();
 editPopup.setEventListeners();
 addPopup.setEventListeners();
+editAvatarPopup.setEventListeners();
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
+editAvatarValidator.enableValidation();
 
 /*--------------------------------------------------*/
 /*                      Functions                   */
 /*--------------------------------------------------*/
 
+function handleDeleteCardClick() {
+  confirmDeleteModal.open();
+  confirmDeleteModal.setSubmitAction(() => {
+    const id = this.getId();
+
+    api
+      .removeCard(id)
+      .then((res) => {
+        this.handleDeleteCard();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+function handleCardLikeClick(cardId, isLiked) {
+  if (isLiked) {
+    api
+      .removeCardLike(cardId)
+      .then((cardData) => this.setLike(cardData.isLiked));
```

Comment:
**Could be improved**

To have access to any method of the class in `index.js` you can pass `this` to the call:

```js

        this._likeButton.addEventListener('click', () => {
            this._handleLike(this);
        });


```

And now in `index.js` you can call the methods like this:


```js


function addLike(cardInstance) {
        api.likeCard(cardInstance.getId())
            .then((res) => {
               cardInstance.addLike(res)
            })
            .catch((err) => {
                console.error(err);
            });
}


```

The same thing is with deleting cards